### PR TITLE
We can RESET a match now!

### PR DESCRIPTION
**Pull request by Jeff-Stammler**
Logger: needed a way to look up if we logged an event (like a starting note).  (explained below)

Match: cleanup of unnecessary things.  Also now look to see if a starting note was logged, rather than look across activities at a static variable (which was a memory leak)

PostMatch: added the reset match bit.

Pre-Match: removed static from checkbox (memory leak).  Always reset the logger (whether from reset match or back button.  Logger constructor does very little now and is safer to do.  Log the startingnote here instead of in Match.

### CODE REVIEW

File: app/src/main/java/com/cpr3663/cpr_scouting_app/Logger.java [hunk 1]
```diff
@@ -338,12 +338,21 @@ public void LogData(String in_Key, String in_Value) {
         match_log_data.add(new Pair<>(in_Key, in_Value.trim()));
     }
 
+    // Member Function: Log a non-time based event - just store this for later.
```
Comment (AdrielStammler) [inline app/src/main/java/com/cpr3663/cpr_scouting_app/Logger.java:341]: This description does not make sense.

File: app/src/main/java/com/cpr3663/cpr_scouting_app/Match.java [hunk 2]
```diff
@@ -529,8 +527,8 @@ public void start_Match() {
         but_MatchControl.setBackgroundColor(getColor(R.color.dark_yellow));
         but_MatchControl.setCompoundDrawablesWithIntrinsicBounds(0, 0, R.drawable.start_teleop, 0);
 
-        if (PreMatch.checkbox_StartNote.isChecked()) {
-            Globals.EventLogger.LogEvent(Constants.EVENT_ID_AUTO_STARTNOTE, 0, 0, true);
+        // If we logged that we started with a note, then set the eventPrevious to it.
+        if (Globals.EventLogger.LookupEvent(Constants.EVENT_ID_AUTO_STARTNOTE)) {
```
Comment (AdrielStammler) [inline app/src/main/java/com/cpr3663/cpr_scouting_app/Match.java:531]: Why don't we log an auto start note?

Comment (AdrielStammler) [reply app/src/main/java/com/cpr3663/cpr_scouting_app/Match.java:531]: Nvm found out why.

**Review by AdrielStammler — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/cpr3663/cpr_scouting_app/Logger.java (hunk 1 above)
Comment (Jeff-Stammler) [reply app/src/main/java/com/cpr3663/cpr_scouting_app/Logger.java:341]: i fixed it in my next upcoming checkin already.

Comment (Jeff-Stammler) [reply app/src/main/java/com/cpr3663/cpr_scouting_app/Logger.java:341]: i fixed it in my next upcoming checkin already.

**Review by AdrielStammler — APPROVED** [review]

**Pull request merged**